src/test/common/clara/test_bindings.cljc

```diff
@@ -369,3 +369,21 @@
              fire-rules
              (query cold-query)))
       "The query results should not be empty for a matching location"))
+
+;; https://github.com/cerner/clara-rules/issues/417
+(def-rules-test test-duplicate-unification
+
+  {:queries [duplcate-unifications [[]
+                                    [[?temp <- Temperature (= ?t temperature)
+                                      (= ?t -10)]]]]
+
+   :sessions [empty-session [duplcate-unifications] {}]}
+
+  (let [session (-> empty-session
+                    (insert (->Temperature -10 "MCI"))
+                    (insert (->Temperature 10 "LAX"))
+                    (insert (->Temperature 30 "ATL"))
+                    fire-rules)]
+    (is (every? (comp #(= -10 %) :temperature :?temp)
```

Comment:
This would pass for an arbitrary number of temperatures; I don't see a reason to be so nonspecific.  I'd expect that any failure of the following would be a bug:
````
(is (= [(->Temperature -10 "MCI")] 
           (map :?temp (query session duplicate-unifications))))
````
I try to write tests to pass on the smallest possible set of outcomes that includes all valid outcomes.  For example, if something caused two ``` (->Temperature -10 "MCI") ``` facts to be put in the query results here it would still be best for this test to fail, even if that isn't what we're specifically trying to test here.